src/main/java/net/snowflake/client/jdbc/SnowflakeBaseResultSet.java

```diff
@@ -30,12 +31,18 @@
   protected SnowflakeResultSetMetaDataV1 resultSetMetaData = null;
   protected Map<String, Object> parameters = new HashMap<>();
   private int fetchSize = 0;
+  protected SFSession session = null;
 
   SnowflakeBaseResultSet(Statement statement) throws SQLException {
     this.statement = statement;
     this.resultSetType = statement.getResultSetType();
     this.resultSetConcurrency = statement.getResultSetConcurrency();
     this.resultSetHoldability = statement.getResultSetHoldability();
+    try {
+      this.session = statement.unwrap(SnowflakeStatementV1.class).connection.getSfSession();
+    } catch (SQLException e) {
```

Comment:
What error can happen here? `statement` is always an instance of `SnowflakeStatementV1.class`?
Add an error to the log here.